Add Glossary of Terms to Understanding Airbyte (#6235)
airbytehq/airbyte: docs/understanding-airbyte/glossary.md

# Glossary of Terms

> ### ETL/ELT

Reviewer: Can we put this in alphabetical order

Author: Yep done.

Stands for **E**xtract, **T**ransform, and **L**oad and **E**xtract, **L**oad, and **T**ransform, respectively.

**Extract**: Retrieve data from a [source](../integrations/sources), which can be an application, database, anything really.

> **Transform**: Clean up the data. This is referred to as [normalization](./basic-normalization.md) in Airbyte and involves [deduplication](./connections/incremental-deduped-history.md), changing data types, formats, and more.

Reviewer 2: can you put Transform 3rd in the list?

**Load**: Move data to your [destination](../integrations/destinations).

### Airbyte CDK

The Airbyte CDK (Connector Development Kit) allows you to create connectors for Sources or Destinations. If your source or destination doesn't exist, you can use the CDK to make the building process a lot easier. It generates all the tests and files you need and all you need to do is write the connector-specific code for your source or destination. We created one in Python which you can check out [here](../connector-development/cdk-python) and the Faros AI team created a Javascript/Typescript one that you can check out [here](../connector-development/cdk-faros-js).

### Full Refresh Sync

> A **Full Refresh Sync** will attempt to retrieve all data from the destination every time a sync is run. Then there are two choices, **Overwrite** and **Append**. **Overwrite** deletes the data in the destination before running the sync and **Append** doesn't.

Reviewer 2: ```suggestion
A **Full Refresh Sync** will attempt to retrieve all data from the source every time a sync is run. Then there are two choices, **Overwrite** and **Append**. **Overwrite** deletes the data in the destination before running the sync and **Append** doesn't.
```

### Incremental Sync

> An **Incremental Sync** will only retrieve new data everytime the a sync occurs. The first sync will always attempt to retrieve all the data. If the [destination supports it](https://discuss.airbyte.io/t/what-destinations-support-the-incremental-deduped-sync-mode/89), you can have your data deduplicated. Simply put, this just means that if you sync an updated version of a record you've already synced, it will remove the old record.

Reviewer 2: ```suggestion
An **Incremental Sync** will only retrieve new data from the source every time the a sync occurs. The first sync will always attempt to retrieve all the data. If the [destination supports it](https://discuss.airbyte.io/t/what-destinations-support-the-incremental-deduped-sync-mode/89), you can have your data deduplicated. Simply put, this just means that if you sync an updated version of a record you've already synced, it will remove the old record.
```

### DAG

> DAG stands for **Directed Acyclic Graph**. It's an overly fancy term originally coined by math graph theorists that just describes a tree-like process. For example, in the following diagram, you start at A and can choose B or C, which then proceed to D and E, respectively. This kind of structure is great for representing workflows and is what tools like [Airflow](https://airflow.apache.org/) use to orchestrate the execution of software based on different cases or states.

Reviewer 2: ```suggestion
DAG stands for **Directed Acyclic Graph**. It's a term originally coined by math graph theorists that describes a tree-like process. For example, in the following diagram, you start at A and can choose B or C, which then proceed to D and E, respectively. This kind of structure is great for representing workflows and is what tools like [Airflow](https://airflow.apache.org/) use to orchestrate the execution of software based on different cases or states.
```

Reviewer 2: you should mention that there can not be a loop inside

Reviewer 2: We want to make things simple but we shouldn't belittle definitions (`overly fancy`)!

![](../.gitbook/assets/glossary_dag_example.png)

### Raw Tables

Airbyte spits out tables with the prefix `_airbyte_raw_`. This is your replicated data, but the prefix indicates that it's not normalized. If you select basic normalization, Airbyte will create renamed versions without the prefix.

### AirbyteCatalog
{% hint style="info" %}
This is only relevant for individuals who want to create a connector.
{% endhint %}

> This refers to how you define the data that you can retrieve from a Source. For example, if you want to retrieve `Account` data from your [Salesforce Source](../integrations/sources/salesforce.md), it needs to be defined clearly so that it can be translated to the destination. Learn more [here](./beginners-guide-to-catalog.md).   

Reviewer 2: you could to introduce the concept of schema. that would make the defintiion clearer.

Author: Makes sense, I've updated it with a definition I think non-technical people can understand too.
